docs/basics-in-depth/dependency_injection.md

```diff
@@ -0,0 +1,215 @@
+---
+id: dependency_injection
+title: Dependency Injection
+sidebar_label: Dependency Injection
+---
+
+# What is Dependency Injection?
+
+Dependency injection is a software design pattern in which a component's dependencies are provided by an external source rather than being hard-coded. This is a very powerful pattern that allows us to decouple our code and make it more testable. It also allows us to have a single source of truth for our dependencies.
+
+---
+
+# What is a service locator?
+
+The service locator a pattern in which an object looks up its dependency from a central `locator` object. This `locator` is responsible for creating and managing the lifecycle of the dependencies, and the object looks them up by the type or the name.
+
+# How does Stacked use dependency injection?
+
+Stacked uses a library called [`get_it`](https://pub.dev/packages/get_it) for dependency injection. get_it is a simple yet powerful dependency injection library for Dart and Flutter.
+
+Let's say we have a `ViewModel` called `AuthViewModel` we want to use that depends on a repository `AuthenticationRepository`.
+
+```dart
+abstract class IAuthenticationRepository {
```

Comment:
We don't teach this pattern in dart. We use the service class directly. From our experience It's rare that polymorphism is required for every class so we skip this boilerplate.